core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -155,6 +158,9 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           // in this block are read when requesting executors
           CoarseGrainedSchedulerBackend.this.synchronized {
             executorDataMap.put(executorId, data)
+            if (currentExecutorIdCounter < Integer.parseInt(executorId)) {
+              currentExecutorIdCounter = Integer.parseInt(executorId)
+            }
```

Comment:
Thank for review it. For my understanding, I don't think we can get the max executor ID in executorDataMap. Because, when AM is failure, all the executor are disconnect and be removed, by this time, as the code in method `CoarseGrainedSchedulerBackend.removeExecutor` show, the executor information in executorDataMap also be removed. 
So, I think the executor information in executorDataMap is not complete. What do you think ?